qemu/tests/watchdog.py

```diff
@@ -89,12 +89,42 @@ def _action_check(test, session, watchdog_action):
         """
         # when watchdog action is pause, shutdown, reset, poweroff
         # the vm session will lost responsive
+
+        def check_guest_reboot(pattern):
+            start_time = time.time()
+            while (time.time() - start_time) < vm.REBOOT_TIMEOUT:
+                if pattern in vm.serial_console.get_output():
```

Comment:
`get_output()` would get all the log during VM running, so `if pattern in vm.serial_console.get_output()` can tell you if the VM was rebooted as you expected.